Zend/zend_compile.c

```diff
@@ -4712,6 +4712,169 @@ static void zend_compile_ns_call(znode *result, znode *name_node, zend_ast *args
 }
 /* }}} */
 
+static zend_op *zend_compile_rope_add(znode *result, uint32_t num, znode *elem_node);
+static zend_op *zend_compile_rope_add_ex(zend_op *opline, znode *result, uint32_t num, znode *elem_node);
+static void zend_compile_rope_finalize(znode *result, uint32_t j, zend_op *init_opline, zend_op *opline);
+
+static zend_result zend_compile_func_sprintf(znode *result, zend_ast_list *args) /* {{{ */
+{
+	if (args->children < 1) {
+		return FAILURE;
+	}
```

Comment:
This took me a while to figure out _why_ this is a FAILURE, it is because we don't have the `$format` string, so maybe add a comment here? Also doesn't seem like you got a test for no args in the optimization tests, or is this unneeded?

Reply:
I find this condition to be the most clear of them all, but added a comment nevertheless. I've also added a corresponding test case, but it's not really necessary: The function bailed out with failure before doing any work whatsoever.